src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatter.cs

```diff
@@ -59,6 +60,7 @@ protected virtual long Schedule(object? obj)
             return id;
         }
 
+        [Obsolete(Obsoletions.InsecureSerializationMessage, DiagnosticId = Obsoletions.InsecureSerializationDiagId, UrlFormat = Obsoletions.SharedUrlFormat)]
```

Comment:
Interesting, it's not just BinaryFormatter but all Formatters.  I hadn't comprehended that.  Not that it matters much probably, since it's rare to see others these days.  And it makes sense they'd all fundamentally have the same issues.

Reply:
I hadn't realized it encompassed IFormatter, Formatter, and their derived types either. Have we researched the extent to which either of these two are implemented outside of our own code - eg a search through NuGet?

Reply:
Right. `Formatter` is already essentially a dead class anyway. It was never really used, not even on Full Framework. `IFormatter` doesn't have much use any more since `NetDataContractSerializer` was never ported over, and there's no remoting infrastructure. So they're both vestigial types at this point.

Reply:
Right, but in theory there could be libraries out there that use either of them (most likely the interface).

@marklio is it possible to construct a query over NuGet to see whether there are libraries out there that implement IFormatter for some reason? It is not in itself dangerous, so in theory we could leave it around, I guess. I am guessing nobody actually uses it.

I think the usage I see in apisof.net is folks accepting/returning it, but that would/could be largely in the service of BinaryFormatter.

Reply:
Not pushing back on obsoleting it otherwise

Reply:
@danmosemsft We don't have the data indexed in a way to ask this question (implemented vs. referenced) with a quick query. If we thought this was important data to make a decision on, I can definitely get this data in the 2-3 days timeframe (or faster if we really think it is important). In any case, I'll add this sort of index to the Chem backlog for the future.

Reply:
@marklio it's not urgent. I don't think it blocks this PR even. probably worth doing, though.

Reply:
> I think the usage I see in apisof.net is folks accepting/returning it, but that would/could be largely in the service of BinaryFormatter.

This matches what I saw as well. `IFormatter` isn't a terribly convenient interface to implement. The consumers of the API that I came across were using it to dynamically switch between `BinaryFormatter`, `NetDataContractSerializer`, and `ObjectStateFormatter`, depending on their scenario. But since those last two don't exist in Core, it means the interface is effectively now synonymous with the one remaining implementation.